### ISSUE TITLE
reconstruction quality on redwood

### ISSUE THREAD

**Issue by yzp12** [IntelVCL/Open3D#465]
I followed the tutorial on reconstruction section and ran reconstruction on 100 models from http://redwood-data.org/3dscan/dataset.html?c=chair. I found the quality is not as good as the reconstructions provided by choi2015. For example, following two pictures are reconstruction by open3d and choi2015 respectively.
![screenshot from 2018-07-20 17-38-49](https://user-images.githubusercontent.com/9149228/43028196-09354da0-8c44-11e8-9cb7-201b5f1dfbea.png)
![screenshot from 2018-07-20 17-39-17](https://user-images.githubusercontent.com/9149228/43028197-0c98938a-8c44-11e8-857c-47f2ed96f09d.png)
Could anyone explain about the discrepancy? I thought open3d should be perform no worse than that as it use more advanced algorithms introduced in Park2017.
Thanks!

**Comment by syncle** [comment]
Oh, interesting. Is that this sequence? http://redwood-data.org/3dscan/dataset.html?c=chair&i=288
I will have a look.

**Comment by syncle** [comment]
Oh, I know why. This dataset is not synchronized. You may notice depth images are 2032 and color images are 2024 frames. I have actually one python script that can do synchronization received from anonymous Open3D user. Let me use it and keep you posted.

**Comment by yzp12** [comment]
Indeed, thank you for doing the test!

**Comment by syncle** [comment]
Hi. This sequence has several issues:
- The sequence has a moving object - there is flag on the scene, and the only moving flag is captured for some time. That challenged RGBD odometry a lot.
- The depth quality is bad - the depth is captured in invalid range (particularly among 1900~2000 frames, almost of the depth pixels are invalid).

I will update my findings to hopefully cover this sequence too.

**Comment by yzp12** [comment]
Hi @syncle ,
You are right, this sequence may indeed be too difficult. But i found similar problems reconstructing other sequences, such like [http://redwood-data.org/3dscan/dataset.html?c=chair&i=21.](url). 
I got something like this:

![screenshot from 2018-06-01 01-54-17](https://user-images.githubusercontent.com/9149228/43174626-c252bc98-8f80-11e8-9209-2d44c93e8099.png)

**Comment by yzp12** [comment]
Hi @syncle ,
You are right, this sequence may indeed be too difficult. But i found similar problems reconstructing other sequences, such like http://redwood-data.org/3dscan/dataset.html?c=chair&i=21..
I got something like this:
![screenshot from 2018-07-24 20-31-51](https://user-images.githubusercontent.com/9149228/43174650-d831dabc-8f80-11e8-951d-d24d937051b9.png)

**Comment by syncle** [comment]
I just tried this sequence, and found some reasons
- RGBD odometry regards the matching process is failure if the number of the correspondence pixels are smaller than ``image_width x image_height x 0.3``. 
- However, the redwood dataset captures the object closely, and most of the pixels are not in the depth camera range. Therefore, it is often the case that valid number of correspondences are smaller than ``image_width x image_height x 0.3``.
- This mechanism blocks to compute real odometry, but accumulates an identity matrix as a estimated odometry. 

I removed this mechanism and got the following result:
<img width="1072" alt="screen shot 2018-07-25 at 8 45 51 am" src="https://user-images.githubusercontent.com/18297113/43218885-978c5d32-8ffa-11e8-9f07-ea5815897232.png">

I will prepare PR to fix this issue and keep investigate possible reasons.
